src/common/commonutils/OtherUtils.c

```diff
@@ -32,42 +32,41 @@ char* DuplicateStringToLowercase(const char* source)
 
 char* FormatAllocateString(const char* format, ...)
 {
-    char* buffer = NULL;
     char* stringToReturn = NULL;
     int formatResult = 0;
-    int sizeOfBuffer = MAX_STRING_LENGTH;
+    int sizeOfBuffer = 0;
 
-    if (NULL == format) 
+    if (NULL == format)
     {
         return stringToReturn;
     }
 
-    while (NULL == stringToReturn)
-    {
-        if (NULL == (buffer = malloc(sizeOfBuffer)))
-        {
-            break;
-        }
-
-        memset(buffer, 0, sizeOfBuffer);
+    va_list arguments;
+    va_start(arguments, format);
+    // snprintf returns the required buffer size, excluding the null terminator
+    sizeOfBuffer = vsnprintf(NULL, 0, format, arguments);
+    va_end(arguments);
 
-        va_list arguments;
-        va_start(arguments, format);
-        formatResult = vsnprintf(buffer, sizeOfBuffer, format, arguments);
-        va_end(arguments);
-
-        if ((formatResult > 0) && (formatResult < (int)sizeOfBuffer))
-        {
-            stringToReturn = DuplicateString(buffer);
-            break;
-        }
+    if (sizeOfBuffer < 0)
+    {
+        return NULL;
+    }
 
-        FREE_MEMORY(buffer);
-        sizeOfBuffer += 1;
+    stringToReturn = malloc((size_t)sizeOfBuffer + 1);
+    if (NULL == stringToReturn)
+    {
+        return NULL;
     }
 
-    FREE_MEMORY(buffer);
+    va_start(arguments, format);
+    formatResult = vsnprintf(stringToReturn, sizeOfBuffer + 1, format, arguments);
+    va_end(arguments);
 
+    if ((formatResult < 0) || (formatResult > sizeOfBuffer))
+    {
+        FREE_MEMORY(stringToReturn);
+        return NULL;
+    }
     return stringToReturn;
 }

```

Comment:
This is a great idea! Just a nit suggestion, same code, just bit changed to only return once after initial input args validation (for consistence with rest of code), how about this?

char* FormatAllocateString(const char* format, ...)
{
    char* stringToReturn = NULL;
    int formatResult = 0;
    int sizeOfBuffer = 0;

    if (NULL == format)
    {
        return stringToReturn;
    }

    va_list arguments;
    va_start(arguments, format);
    // snprintf returns the required buffer size, excluding the null terminator
    sizeOfBuffer = vsnprintf(NULL, 0, format, arguments);
    va_end(arguments);

    if (sizeOfBuffer > 0)
    {
        if (NULL != (stringToReturn = malloc((size_t)sizeOfBuffer + 1)))
        {
            va_start(arguments, format);
            formatResult = vsnprintf(stringToReturn, sizeOfBuffer + 1, format, arguments);
            va_end(arguments);

            if ((formatResult < 0) || (formatResult > sizeOfBuffer))
            {
                FREE_MEMORY(stringToReturn);
            }
        }
    }

    return stringToReturn;
}

Reply:
Restructured to match the structure you suggested, with one difference: `if (sizeOfBuffer >= 0)` instead of `if (sizeOfBuffer > 0)`. This is so that we don't return NULL on an invocation with valid inputs (for example `FormatAllocateString("%s", "")`). There is an added test case for this.

Reply:
Thanks!